elements/pf-back-to-top/test/pf-back-to-top.spec.ts

```diff
@@ -0,0 +1,79 @@
+import { expect, html, nextFrame, fixture } from '@open-wc/testing';
+import { createFixture } from '@patternfly/pfe-tools/test/create-fixture.js';
+import { setViewport } from '@web/test-runner-commands';
+
+import { allUpdates } from '@patternfly/pfe-tools/test/utils.js';
+
+import { PfBackToTop } from '../pf-back-to-top.js';
+
+describe('<pf-back-to-top>', function() {
+  let element: PfBackToTop;
+  let link: HTMLAnchorElement;
+
+  beforeEach(async function() {
+    await setViewport({ width: 320, height: 640 });
+    window.scrollTo({ top: 0, behavior: 'instant' });
+    await nextFrame();
+    await fixture<HTMLElement>(html`<div style="height: 2000px;"></div>`);
+    element = await createFixture<PfBackToTop>(html`<pf-back-to-top>Back to Top</pf-back-to-top>`);
+    await allUpdates(element);
+  });
+
+  it('imperatively instantiates', function() {
+    expect(document.createElement('pf-back-to-top')).to.be.an.instanceof(PfBackToTop);
+  });
+
+  it('should upgrade', async function() {
+    const klass = customElements.get('pf-back-to-top');
+    expect(element)
+      .to.be.an.instanceOf(klass)
+      .and
+      .to.be.an.instanceOf(PfBackToTop);
+  });
+
+  describe('with `href` attribute', function() {
+    beforeEach(async function() {
+      element.setAttribute('href', '#top');
+      await allUpdates(element);
+      link = element.shadowRoot!.querySelector('a')!;
+    });
+
+    it('should be visible after 400px scroll', async function() {
+      expect(link).to.have.class('visually-hidden');
+      window.scrollTo({ top: 401, behavior: 'instant' });
+      await allUpdates(element);
+      await nextFrame();
+      expect(link).to.not.have.class('visually-hidden');
+    });
```

Comment:
```suggestion
    it('should be hidden', async function() {
      const snapshot = await a11ySnapshot();
      expect(snapshot).to.deep.equal(/* expected */);
    });
    
    describe('after 400px scroll', function() {
      beforeEach(async function() {
        window.scrollTo({ top: 401, behavior: 'instant' });
        await allUpdates(element);
        await nextFrame();
      });
          
      it('should be visible', async function() {
        const snapshot = await a11ySnapshot();
        expect(snapshot).to.deep.equal(/* expected */);
      });
    });

```

etc, in other places. don't assert on shadow dom, instead assert on the ax tree. `it(title, async function() {});` is a smell, the `async` should not be necessary. we can make an exception for asserting on the ax tree because that's an async function, but in generally we should seek to find async functions mostly in beforeEach blocks only.

Reply:
I'm not modifying the ally tree here though, so not sure what to assert on there.  Its not `hidden` its `visually-hidden` off screen.  This leaves the link available to screen readers at all times since they don't "scroll".  Maybe need to reevaluate that from an accessibility standpoint /cc @nikkimk

Reply:
what happens when you focus it?

Reply:
Point taken, large refactor of the component done along with tests, PTAL.